### ISSUE TITLE
Tab-completion doesn't work with absolute paths

### ISSUE THREAD

**Issue by posita** [Trudko/advanced-new-file#18]
To reproduce (at least on OS X):

* Open the new file dialog
* Type the absolute path to your home directory (e.g., `/Users/posita`, `/home/matt`, etc.)
* Hit tab

No files or directories are listed.

**Comment by posita** [comment]
Actually, I think I understand better what's going on. You can see it here (set to loop twice):

![foo](https://cloud.githubusercontent.com/assets/222581/7260245/9dc8e0d6-e81d-11e4-9e1b-dae9eb0747ae.gif)

Basically, I hit tab after `b` matches `bin` in the local directory, and the relative path is replaced with what *appears* to be an absolute path. *However*, absolute paths are still being treated as relative (i.e., `/Users/matt + /Users/matt = /Users/matt/Users/matt`). Compare this with, e.g., Python's approach, where the second path will overwrite the first if it's an absolute path:

```py
% python
...
>>> import os.path
>>> os.path.join('/Users/posita', '/home/matt')
'/home/matt'
```

Node seems to have some [helpful (similar) APIs](https://nodejs.org/api/path.html), but I don't know how portable they are. There might be others, but I'm not familiar with the options. Perhaps Atom has some built-ins? I [didn't see any](https://atom.io/docs/api/search/latest?utf8=%E2%9C%93&q=path) that jumped out at me.

**Comment by Trudko** [comment]
Seems like bug, I am not able to reproduce it yet. Need bit more time for this.

**Comment by Trudko** [comment]
I was able to reproduce it if I try to wrote absolute path - /home/Trudko/Desktop/file.txt for example. This was create file under /home/Trudko/home/Trudko/Desktop but that make sense because package  creates file relatively to project folder. 

How exactly do you use this package? Do you open home folder in Atom? Feels like you are trying to create files outside of folder/project being open right now in Atom. Am I correct?

**Comment by posita** [comment]
> How exactly do you use this package? Do you open home folder in Atom? Feels like you are trying to create files outside of folder/project being open right now in Atom. Am I correct?

Possibly, but there may not even be a project open (i.e., the "project" could be the user's home directory), and this plugin is still useful. There are all kinds of reasons why one might want to create a new file outside of the current project tree.

**Comment by eboracus** [comment]
Yes I agree, I think it would be optimal if this package could generate new files outside as well as inside (which should still be default) the current project tree. Maybe with a toggle automatically thrown (and displayed in the dialog) when an absolute path from the home directory is entered, this could then be switched off if the user actually wanted a coincidentally named dub-directory within the current project.

The loophole in the current behaviour being that I came upon this issue trying to navigate up the directory hierarchy using "../" and tab complete. Tabbing out the known absolute path folder by folder worked successfully, showing all known folders as expected on the way, leading to the impression that the new file would be created in the absolute path, but this still created a sub-directory in the current project tree as discussed above.

![advancednewfile_absolute_paths](https://cloud.githubusercontent.com/assets/11758299/8467751/c00da436-205a-11e5-9906-e003acc46295.png)
Navigating out of the current project root... I'm thinking a visual toggle/flag somewhere in there to show you are now going to create an absolute path to the file (and which stays even when the "../../" disappears on tab completion press) would be a good solution. And toggle off to use relative path.

**Comment by posita** [comment]
Closed as `wontfix`?
